src/utils/ContractUtils.ts

```diff
@@ -162,3 +162,29 @@ export async function amountToUnits(
     LoggerInstance.error(`ERROR: FAILED TO CALL DECIMALS(), USING 18', ${e.message}`)
   }
 }
+
+/**
+ * Estimates the gas used when a function would be executed on chain
+ * @param {string} from account that calls the function
+ * @param {Function} functionToEstimateGas function that we need to estimate the gas
+ * @param {...any[]} args arguments of the function
+ * @return {Promise<number>} gas cost of the function
+ */
+export async function estimateGas(
```

Comment:
Can we somehow write a test for this method ? I'm thinking of having picking one contract method for eg and compare the value returned by this method to the value returned by calling estimate directly on the contract method as we did before ? Or maybe something else if you have a better idea .
By doing this i think we may also not have issues with code climate coverage dropping and the diff coverage